pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -151,6 +154,20 @@ cdef dict timedelta_abbrevs = {
 
 _no_input = object()
 
+# pytimedeltas may fall in the following range: [-999999999 days, 1000000000 days)
+# for example, 999999999 days, 86399 seconds is in the range,
+# whereas -999999999 days, -86399 seconds is not in the range
+
+# upper bound for unit seconds: 1000000000 days * 86400 s/day
+cdef int64_t PYTIMEDELTA_UPPER_S = 86400000000000
```

Comment:
are these importable from Python.h? Or where did these come from?

Reply:
Thanks for pointing that out -- found the constant in the Python datetime module today and just made a new commit that uses it. Left comments and examples as they are as I think it's illustrative to show how large these bounds are. Let me know if you have further thoughts on how this was done.

Reply:
Realized I'm doing this in the wrong way...trying a couple of alternatives now to import from CPython headers.

Reply:
Finally realized that the constant is _not_ defined in datetime.h, but in some other c file in CPython. Defined it in the exact same way in this pandas file, but we can always squash this revision to just use magic numbers as in the first set of commits.

If there's a way I could import the constant from that c file in CPython, do let me know! I just can't think of or find a clear cut way to do that.